selfservice/hook/show_verification_ui.go

```diff
@@ -0,0 +1,61 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package hook
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ory/kratos/driver/config"
+	"github.com/ory/kratos/selfservice/flow"
+	"github.com/ory/kratos/selfservice/flow/registration"
+	"github.com/ory/kratos/session"
+	"github.com/ory/kratos/x"
+	"github.com/ory/x/otelx"
+)
+
+var (
+	_ registration.PostHookPostPersistExecutor = new(SessionIssuer)
+)
+
+type (
+	showVerificationUIDependencies interface {
+		x.WriterProvider
+		config.Provider
+	}
+
+	ShowVerfificationUIProvider interface {
+		HookShowVerificationUI() *ShowVerificationUI
+	}
+
+	ShowVerificationUI struct {
+		d showVerificationUIDependencies
+	}
+)
+
+func NewShowVerificationUI(d showVerificationUIDependencies) *ShowVerificationUI {
+	return &ShowVerificationUI{d: d}
+}
+
+func (e *ShowVerificationUI) ExecutePostRegistrationPostPersistHook(w http.ResponseWriter, r *http.Request, f *registration.Flow, s *session.Session) error {
+	return otelx.WithSpan(r.Context(), "selfservice.hook.SessionIssuer.ExecutePostRegistrationPostPersistHook", func(ctx context.Context) error {
+		return e.execute(w, r.WithContext(ctx), f, s)
+	})
+}
+
+func (e *ShowVerificationUI) execute(w http.ResponseWriter, r *http.Request, f *registration.Flow, s *session.Session) error {
+	var vf *flow.ContinueWithVerificationUI
+	for _, c := range f.ContinueWithItems {
+		if item, ok := c.(*flow.ContinueWithVerificationUI); ok {
+			vf = item
+		}
+	}
+
+	if vf != nil {
+		redir := e.d.Config().SelfServiceFlowVerificationUI(r.Context())
+		x.ContentNegotiationRedirection(w, r, s.Declassified(), e.d.Writer(), vf.AppendTo(redir).String())
```

Comment:
Does this work for native apps?

Reply:
Good point, I've added a check to only execute this for browsers. Should be fine like that.

Reply:
It doesn't, that was my point on Friday. This hook can only be used with server rendered browser flows. 

For native apps you get the `continue_with` array in the JSON response.

Reply:
Got it!